models/gc_spider_baseline/utils/SpiderBaselineRunner.py

```diff
@@ -0,0 +1,195 @@
+import json
+
+import SimpleITK
+import numpy as np
+
+from mhubio.core import Instance, InstanceData, IO, Module, CT, MR, Meta
+
+# The GC Spider baseline algorithm is imported as a self-contained algorithm class with an execute method
+from TestSpine import Module as SpiderAlgorithm
+
+
+SPIDER_DATA_DIR = Path(os.environ["VERSEG_BASEDIR"])
+SPIDER_INTERNAL_DATASET_PATH = SPIDER_DATA_DIR / "datasets" / "spider_input"
+SPIDER_EXPERIMENT_DIR = SPIDER_DATA_DIR / "experiments" / "SPIDER-Baseline"
+SPIDER_INTERNAL_OUTPUT_DIR = SPIDER_EXPERIMENT_DIR / "results" / "spider_input"
+SPIDER_EXPERIMENT_ARGUMENTS_FILE = SPIDER_EXPERIMENT_DIR / "arguments.json"
+SPIDER_INTERNAL_DATASET_META_FILE = SPIDER_INTERNAL_DATASET_PATH / "metadata.json"
+
+
+@IO.Config('traversal_direction_up', bool, True, the="Direction to traverse the image")
+@IO.ConfigInput('in_data', 'mha:mod=ct|mr', the='supported datatypes for the spider baseline model')
+class SpiderBaselineRunner(Module):
+
+    traversal_direction_up: bool
+
+    @IO.Instance()
+    @IO.Input('in_data', the='input sagittal spine image (MR or CT)')
+    @IO.Output(
+        'out_data_raw',
+        'spider_baseline_vertebrae_segmentation_raw.mha',
+        'mha:mod=seg:model=SpiderBaselineSegmentation:seg=raw',
+        data='in_data',
+        bundle='model',
+        the='Spider baseline vertebrae segmentation for the input sagittal spine MRI or CT. Raw segmentation output: '
+            '0: background '
+            '1-24: different vertebrae numbered from the bottom (i.e. L5 = 1, C1 = 24) '
+            '100: spinal canal '
+            '101-124 for partially visible vertebrae '
+            '201-224: different intervertebral discs (also numbered from the bottom, i.e. L5/S1 = 201, C2/C3 = 224) '
+    )
+    @IO.Output(
+        'out_data',
+        'spider_baseline_vertebrae_segmentation.mha',
+        'mha:mod=seg:model=SpiderBaselineSegmentation:seg=remapped:roi=VERTEBRAE_L5,VERTEBRAE_L4,VERTEBRAE_L3,VERTEBRAE_L2,'
+        'VERTEBRAE_L1,VERTEBRAE_T12,VERTEBRAE_T11,VERTEBRAE_T10,VERTEBRAE_T9,VERTEBRAE_T8,VERTEBRAE_T7,VERTEBRAE_T6,VERTEBRAE_T5,'
+        'VERTEBRAE_T4,VERTEBRAE_T3,VERTEBRAE_T2,VERTEBRAE_T1,VERTEBRAE_C7,VERTEBRAE_C6,VERTEBRAE_C5,VERTEBRAE_C4,VERTEBRAE_C3,VERTEBRAE_C2,VERTEBRAE_C1,'
+        'VERTEBRAE_DISK_L5S1,VERTEBRAE_DISK_L4L5,VERTEBRAE_DISK_L3L4,VERTEBRAE_DISK_L2L3,VERTEBRAE_DISK_L1L2,VERTEBRAE_DISK_T12L1,VERTEBRAE_DISK_T11T12,'
+        'VERTEBRAE_DISK_T10T11,VERTEBRAE_DISK_T9T10,VERTEBRAE_DISK_T8T9,VERTEBRAE_DISK_T7T8,VERTEBRAE_DISK_T6T7,VERTEBRAE_DISK_T5T6,VERTEBRAE_DISK_T4T5,'
+        'VERTEBRAE_DISK_T3T4,VERTEBRAE_DISK_T2T3,VERTEBRAE_DISK_T1T2,VERTEBRAE_DISK_C7T1,VERTEBRAE_DISK_C6C7,VERTEBRAE_DISK_C5C6,VERTEBRAE_DISK_C4C5,'
+        'VERTEBRAE_DISK_C3C4,VERTEBRAE_DISK_C2C3,'
+        'SPINAL_CANAL',
+        data='in_data',
+        bundle='model',
+        the='Spider baseline vertebrae segmentation for the input sagittal spine MRI or CT. Remapped segmentation output: '
```

Comment:
For better readability, this can go into the model card description (it's redundant here as we now have all SegDB IDs in the roi metadata).

Reply:
I think you are referring to the index descriptions under `the`? I kind of understand your argument and I agree that it should be updated in the model card description, but it seems still nice to have a more readable summary of the rois section next to the roi list since large lists of roi strings are hard to read for users/developers.

Reply:
I'm fine with both. To give you some context: I introduced the decorators not only to provide a standardized framework and increase readability but also because their easily parsable and will allow automatic generation of documentation for all (custom) MHub-IO modules in the future.